### PR TITLE
New Code: adding js assignment 6 string and string methods

### PR DESCRIPTION
**Pull request by poonam-khode**

### CODE REVIEW

**Review by MKadeer17 — APPROVED** [review]

File: poonam/JS/Assignment6_js/string.html [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            <h4 id="res2"></h4>
+
+            <h3>Last index</h3>
+            <label for="val3">Enter text:</label>
+            <input type="text" name="val3" id="val3"><br><br>
+            <label for="ind2">Find index of:</label>
+            <input type="text" id="ind2"><br>
+            <button type="button" onclick="lastindex()">Submit</button>
+            <h4 id="res3"></h4>
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add comment this line

File: poonam/JS/Assignment6_js/string.html [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        </script>
+    </head>
+    <body>
+        <form action="">
+            <h3>Trim string</h3>
+            <label for="val">Enter text:</label>
+            <input type="text" name="val" id="val"><br>
+            <button type="button" onclick="trim()">Submit</button>
+            <h4 id="res"></h4>
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            <h3>String length</h3>
+            <label for="val1">Enter text:</label>
+            <input type="text" name="val1" id="val1"><br>
+            <button type="button" onclick="strlength()">Submit</button>
+            <h4 id="res1"></h4>
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            <h3>Index</h3>
+            <label for="val2">Enter text:</label>
+            <input type="text" name="val2" id="val2"><br><br>
+            <label for="ind1">Find index of:</label>
+            <input type="text" id="ind1"><br>
+            <button type="button" onclick="index()">Submit</button>
+            <h4 id="res2"></h4>
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+            <h4 id="res3"></h4>
+
+            <h3>Concat</h3>
+            <label for="val4">Enter text:</label>
+            <input type="text" name="val4" id="val4"><br><br>
+            <label for="val5">First index:</label>
+            <input type="text" name="val5" id="val5"><br>
+            <button type="button" onclick="concatstr()">Submit</button>
+            <h4 id="res4"></h4>
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            <h3>Slice</h3>
+            <label for="val6">Enter text:</label>
+            <input type="text" name="val6" id="val6"><br><br>
+            <label for="find">First index:</label>
+            <input type="number" id="find"><br><br>
+            <label for="lind">Last index:</label>
+            <input type="number" id="lind"><br>
+            <button type="button" onclick="slicestr()">Submit</button>
+            <h4 id="res5"></h4>
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+            <h3>Lower case</h3>
+            <label for="val7">Enter text:</label>
+            <input type="text" name="" id="val7"><br>
+            <button type="button" onclick="lower_case()">Submit</button>
+            <h4 id="res6"></h4>
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+            <h3>Upper case</h3>
+            <label for="val8">Enter text:</label>
+            <input type="text" name="" id="val8"><br>
+            <button type="button" onclick="upper_case()">Submit</button>
+            <h4 id="res7"></h4>
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+            <h3>Search string</h3>
+            <label for="val9">Enter text:</label>
+            <input type="text" name="val9" id="val9"><br><br>
+            <label for="word">Enter word:</label>
+            <input type="text" name="word" id="word"><br>
+            <button type="button" onclick="search_str()">Submit</button>
+            <h4 id="res8"></h4>
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <title>Document</title>
+        <script>
+            function trim(){
+                var str = document.getElementById("val").value;
+                var output = str.trim();
+                output = document.getElementById("res").innerHTML= "Output = "+output;
+            }
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+            function strlength(){
+                var str1 = document.getElementById("val1").value;
+                var output1 = str1.length;
+                output1 = document.getElementById("res1").innerHTML= "Output = "+output1;
+            }
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+            function index(){
+                var str2 = document.getElementById("val2").value;
+                var ind1 = document.getElementById("ind1").value;
+                var output2 = str2.indexOf(ind1);
+                output2 = document.getElementById("res2").innerHTML= "Output = "+output2;
+            }
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+            function lastindex(){
+                var str3 = document.getElementById("val3").value;
+                var ind2 = document.getElementById("ind2").value;
+                var output3 = str3.lastIndexOf(ind2);
+                output3 = document.getElementById("res3").innerHTML= "Output = "+output3;
+            }
+
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: remove extra line ya add this line comment

File: poonam/JS/Assignment6_js/string.html [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+                var str9 = document.getElementById("val10").value;
+                var nword = document.getElementById("nword").value;
+                var rword = document.getElementById("rword").value;
+                var output9 = str9.replace(nword, rword);
+                output9 = document.getElementById("res9").innerHTML= "Output = "+output9;
+            }
+        </script>
+    </head>
+    <body>
+        <form action="">
```
Comment (VIP5614) [inline poonam/JS/Assignment6_js/string.html]: fill the action attribute please like (#)

**Review by VIP5614 — APPROVED** [review]

File: poonam/JS/Assignment6_js/string.html (hunk 14 above)
Comment (poonam-khode) [reply poonam/JS/Assignment6_js/string.html]: done

**Review by VIP5614 — APPROVED** [review]

File: poonam/JS/Assignment6_js/string.html [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+            function slicestr(){
+                var str5 = document.getElementById("val6").value;
+                var ind1 = document.getElementById("find").value;
+                var ind2 = document.getElementById("lind").value;
+                var output5 = str5.slice(ind1, ind2);
+                output5 = document.getElementById("res5").innerHTML= "Output = "+output5;
+            }
+            //lower case
+            function lower_case(){
+                var str6 = document.getElementById("val7").value;
```
Comment (SnehaMakade) [inline poonam/JS/Assignment6_js/string.html]: change all variable names like string_six and output_six  in all functions.

File: poonam/JS/Assignment6_js/string.html [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+                var output6 = str6.toLowerCase();
+                output6 = document.getElementById("res6").innerHTML= "Output = "+output6;
+            }
+            //upper case
+            function upper_case(){
+                var str7 = document.getElementById("val8").value;
```
Comment (SnehaMakade) [inline poonam/JS/Assignment6_js/string.html]: do not add numbers in variable names instead use word for that like string_seven, varibale name should be identical

File: poonam/JS/Assignment6_js/string.html [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+            <label for="val2">Enter text:</label>
+            <input type="text" name="val2" id="val2"><br><br>
+            <label for="ind1">Find index of:</label>
+            <input type="text" id="ind1"><br>
+            <button type="button" onclick="index()">Submit</button>
+            <h4 id="res2"></h4>
+            <h3>Last index</h3>
+            <label for="val3">Enter text:</label>
+            <input type="text" name="val3" id="val3"><br><br>
+            <label for="ind2">Find index of:</label>
```
Comment (SnehaMakade) [inline poonam/JS/Assignment6_js/string.html]: change label as this is for finding last index of word or string

File: poonam/JS/Assignment6_js/string.html [hunk 18]
```diff
@@ -0,0 +1,149 @@
...
+            <input type="text" id="ind2"><br>
+            <button type="button" onclick="lastindex()">Submit</button>
+            <h4 id="res3"></h4>
+            <h3>Concat</h3>
+            <label for="val4">Enter text:</label>
+            <input type="text" name="val4" id="val4"><br><br>
+            <label for="val5">First index:</label>
```
Comment (SnehaMakade) [inline poonam/JS/Assignment6_js/string.html]: label should be different here as its for concat strings like enter second text, so give appropriate labels everywhere.

File: poonam/JS/Assignment6_js/string.html [hunk 19]
```diff
@@ -0,0 +1,149 @@
...
+            <button type="button" onclick="lower_case()">Submit</button>
+            <h4 id="res6"></h4>
+            <h3>Upper case</h3>
+            <label for="val8">Enter text:</label>
+            <input type="text" name="val8" id="val8"><br>
+            <button type="button" onclick="upper_case()">Submit</button>
+            <h4 id="res7"></h4>
+            <h3>Search string</h3>
+            <label for="val9">Enter text:</label>
+            <input type="text" name="val9" id="val9"><br><br>
```
Comment (SnehaMakade) [inline poonam/JS/Assignment6_js/string.html]: give proper id names in all input box

File: poonam/JS/Assignment6_js/string.html [hunk 20]
```diff
@@ -0,0 +1,149 @@
...
+            <input type="text" name="val10" id="val10"><br><br>
+            <label for="nword">Enter word:</label>
+            <input type="text" name="nword" id="nword"><br><br>
+            <label for="rword">Enter replace word:</label>
+            <input type="text" name="rword" id="rword"><br>
+            <button type="button" onclick="replace_str()">Submit</button>
+            <h4 id="res9"></h4>
+        </form>
+    </body>
+</html>
```
Comment (SnehaMakade) [inline poonam/JS/Assignment6_js/string.html:159]: add new end line

**Review by SnehaMakade — CHANGES_REQUESTED** [review]
Do changes as per comments given, like change id, label and variable names as per their purpose.

**Review by sanketmod — APPROVED** [review]

**Pull request merged**